dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/AssemblyCheckFilterProcessor.java

```diff
@@ -64,17 +64,27 @@ public SubSnpCoreFields process(SubSnpCoreFields subSnpCoreFields) throws Except
     private boolean referenceAlleleIsCorrect(String referenceAllele, SubSnpCoreFields subSnpCoreFields) {
 
         Region region = subSnpCoreFields.getVariantCoordinates();
+        String contigName = subSnpCoreFields.getContigRegion().getChromosome();
         try {
-            String sequenceInAssembly = fastaReader.getSequence(region.getChromosome(), region.getStart(),
-                                                                region.getEnd());
-            if (referenceAllele.equals(sequenceInAssembly)) {
+            if (referenceAllele.isEmpty()) {
+                if (!fastaReader.doesContigExist(contigName)) {
+                    throw new IllegalArgumentException("Sequence " + contigName + " not found in reference fasta file");
+                }
                 return true;
             } else {
-                logger.warn(
-                        "Variant filtered out because the reference allele does not match the reference sequence: {}" +
-                                ".\n{} expected in {}, {} found",
-                        subSnpCoreFields, sequenceInAssembly, region, referenceAllele);
-                return false;
+                String sequenceInAssembly = fastaReader.getSequence(region.getChromosome(), region.getStart(),
+                                                                    region.getEnd());
+                if (referenceAllele.equals(sequenceInAssembly)) {
+                    return true;
+                } else {
+                    logger.warn(
+                            "Variant filtered out because the reference allele does not match the reference sequence: {}" +
+
+
+                                    ".\n{} expected in {}, {} found",
```

Comment:
I had to deal with this message in the logs, and it's not really clear what's the reference sequence and what's the allele in the variant. maybe "Reference sequence is {} in region {}, the variant had {}".